tests/Gelf/Test/Transport/RetryTransportWrapperTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+
+namespace Gelf\Test\Transport;
+
+
+use Gelf\Message;
+use Gelf\TestCase;
+use Gelf\Transport\AbstractTransport;
+use Gelf\Transport\RetryTransportWrapper;
+use \PHPUnit_Framework_MockObject_MockObject as MockObject;
+
+class RetryTransportWrapperTest extends TestCase
+{
+    /**
+     * @var Message
+     */
+    private $message;
+
+    /**
+     * @var AbstractTransport|MockObject
+     */
+    private $transport;
+
+    /**
+     * @var RetryTransportWrapper
+     */
+    private $wrapper;
+
+    /**
+     * @var string
+     */
+    private $successValue;
+
+    public function setUp()
+    {
+        $this->message = new Message();
+        $this->transport = $this->buildTransport();
+        $this->wrapper   = new RetryTransportWrapper($this->transport);
+        $this->successValue = "HTTP/1.1 202 Accepted\r\n\r\n";
+    }
+
+    public function testSendFailOnce()
+    {
+        $expectedException = new \RuntimeException();
+
+        $this->transport->expects($this->exactly(2))
+            ->method('send')
+            ->with($this->message)
+            ->will($this->onConsecutiveCalls(
+                $this->throwException($expectedException),
+                $this->returnValue($this->successValue)
+            ));
+
+        $bytes = $this->wrapper->send($this->message);
+
+        $this->assertEquals($this->successValue, $bytes);
+    }
+
+    public function testSendFail()
+    {
+        $expectedException = new \RuntimeException();
+
+        $this->transport->expects($this->exactly(2))
+            ->method('send')
+            ->with($this->message)
+            ->will($this->onConsecutiveCalls(
+                $this->throwException($expectedException)));
```

Comment:
Since you're able to perform an `assertNotEquals` assertion, I think the second call will return a `NULL`. Thus, you're not testing what you're expecting. To truly test that both calls fails, you should update `onConsecutiveCalls` like this:

```suggestion
            ->will($this->onConsecutiveCalls(
                $this->throwException($expectedException1),
                $this->throwException($expectedException2),
            ));
```

Then, you can add the following:

```php
$this->expectException(RuntimeException::class);
```

You might also want to add different error messages to your exception and add the following assertion:

```php
$this-expectExceptionMessage('Exception 2');
```